pkg/resourceinterpreter/configurableinterpreter/configurable.go

```diff
@@ -2,142 +2,212 @@ package configurableinterpreter
+	"k8s.io/client-go/tools/cache"
 	"k8s.io/klog/v2"
 
 	configv1alpha1 "github.com/karmada-io/karmada/pkg/apis/config/v1alpha1"
 	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
-	"github.com/karmada-io/karmada/pkg/resourceinterpreter/configurableinterpreter/configmanager"
+	"github.com/karmada-io/karmada/pkg/resourceinterpreter/configurableinterpreter/engine"
+	"github.com/karmada-io/karmada/pkg/resourceinterpreter/configurableinterpreter/engine/lua"
 	"github.com/karmada-io/karmada/pkg/util/fedinformer/genericmanager"
+	"github.com/karmada-io/karmada/pkg/util/helper"
 )
 
+var resourceInterpreterCustomizationsGVR = schema.GroupVersionResource{
+	Group:    configv1alpha1.GroupVersion.Group,
+	Version:  configv1alpha1.GroupVersion.Version,
+	Resource: "resourceinterpretercustomizations",
+}
+
 // ConfigurableInterpreter interpret custom resource with resource interpreter customizations configuration
 type ConfigurableInterpreter struct {
-	// configManager caches all re configurations.
-	configManager configmanager.ConfigManager
+	lock         sync.RWMutex
+	interpreters map[schema.GroupVersionKind]map[configv1alpha1.InterpreterOperation]engine.Function
+	configLister cache.GenericLister
 }
 
 // NewConfigurableInterpreter return a new ConfigurableInterpreter.
-func NewConfigurableInterpreter(informer genericmanager.SingleClusterInformerManager) (*ConfigurableInterpreter, error) {
-	return &ConfigurableInterpreter{
-		configManager: configmanager.NewInterpreterConfigManager(informer),
-	}, nil
-}
-
-// HookEnabled tells if any hook exist for specific resource gvk and operation type.
-func (c *ConfigurableInterpreter) HookEnabled(kind schema.GroupVersionKind, operationType configv1alpha1.InterpreterOperation) bool {
-	if !c.configManager.HasSynced() {
-		klog.Errorf("not yet ready to handle request")
-		return false
+func NewConfigurableInterpreter(informer genericmanager.SingleClusterInformerManager) *ConfigurableInterpreter {
+	c := &ConfigurableInterpreter{
+		configLister: informer.Lister(resourceInterpreterCustomizationsGVR),
 	}
-	_, exist := c.configManager.LuaScriptAccessors()[kind]
-	return exist
+
+	informer.ForResource(resourceInterpreterCustomizationsGVR, cache.ResourceEventHandlerFuncs{
+		AddFunc:    func(_ interface{}) { c.updateConfiguration() },
+		UpdateFunc: func(_, _ interface{}) { c.updateConfiguration() },
+		DeleteFunc: func(_ interface{}) { c.updateConfiguration() },
+	})
+	return c
 }
 
-// CheckConfigurableExists check resource interpreter customizations configuration for gkv exist
-func (c *ConfigurableInterpreter) CheckConfigurableExists(gvk schema.GroupVersionKind) bool {
-	_, ok := c.configManager.LuaScriptAccessors()[gvk]
-	return ok
+// GetInterpreter returns interpreter for specific resource kind and operation.
+// Why don't support `HookEnabled` function?
+//    When `HookEnabled` tells true, while when methods are calling, configure may be removed, and panic.
```

Comment:
Because of the `HookEnabled` interface:
https://github.com/karmada-io/karmada/blob/7c8f237d299ad4b6062d14317b6894ff1aa49faa/pkg/resourceinterpreter/interpreter.go#L25
there is always a small time gap between the `HookEnabled` call and the actual method call(such as `GetReplicas`), so we need to reconfirm in the method call, like this:
https://github.com/karmada-io/karmada/blob/7c8f237d299ad4b6062d14317b6894ff1aa49faa/pkg/resourceinterpreter/interpreter.go#L100-L102

`HookEnabled` function is not necessary for `ConfigurableInterpreter`, we can choose not to support it. 

Or I think we can add one more function like:
```go
func (c *ConfigurableInterpreter) HookEnabled(objGVK schema.GroupVersionKind, operation configv1alpha1.InterpreterOperation) bool {
	_, enabled := c.GetInterpreter(objGVK, operation)
	return enabled
}
```

Reply:
ok